spec.emu

```diff
@@ -267,6 +267,8 @@ contributors: Ujjwal Sharma, Younies Mahmoud
             1. Throw a *RangeError* exception.
           1. Else if _unit_ is *"minutes"* or *"seconds"*, then
             1. Set _style_ to *"2-digit"*.
+          1. If _prevStyle_ is *"numeric"* and _style_ is *"numeric"* and _display_ is *"always"* and _unit_ is one of "milliseconds", "microseconds", or "nanoseconds", then
```

Comment:
I think you only need

`If _style_ is *"numeric"* and _display_ is *"always"*`

Why do you need the other 2 conditions?

Reply:
I guess the error is desirable only when dealing with units that show up in the fractional part.

Reply:
> I guess the error is desirable only when dealing with units that show up in the fractional part.

Exactly, because they are the ones whose display is handled entirely independently.

> Why do you need the other 2 conditions?

You're both right. According to https://github.com/tc39/proposal-intl-duration-format/pull/148/files#diff-7d681727fcf47dc0b9a7512a470fb0da63276c625891a5cc232d725bd12912fdR265-R267, if `prevStyle` is `"numeric"`, it implies the current style is `"numeric"` too. We should definitely simplify the condition here.

Reply:
> should they throw RangeError? why?

Good point. It should be RangeError only for numerics that fall in the fractional part, which is ms/μs/ns.

> A. seconds is NOT "numeric" but milliseconds IS "numeric"
> B. seconds AND milliseconds are NOT "numeric" but microseconds IS "numeric"
> C. seconds AND milliseconds AND microseconds are NOT "numeric" but nanoseconds IS "numeric"
>
> what should the format of {seconds: 1, millisconds: 234, microseconds: 567, nanoseconds: 891} be?

A = "1.234567891 sec" because "numeric" for milliseconds and below means to append as a fraction to the previous unit

B = "1 sec and 234.567.891 ms"

C = "1 sec, 234 ms, and 567.891 μs"

Reply:
ok. That make sense.